nameko/timer.py

```diff
@@ -49,26 +52,29 @@ def kill(self):
 
     def _run(self):
         """ Runs the interval loop. """
-
-        sleep_time = self.interval
+        sleep_time = 0 if self.eager else self.interval
 
         while True:
             # sleep for `sleep_time`, unless `should_stop` fires, in which
             # case we leave the while loop and stop entirely
-            with Timeout(sleep_time, exception=False):
-                self.should_stop.wait()
-                break
+            time.sleep(sleep_time)
 
             start = time.time()
 
             self.handle_timer_tick()
 
+            self.worker_complete.wait()
```

Comment:
at this point an eventlet `Semaphore` (remember to init with zero, not the default 1) might be more idiomatic